### PR TITLE
Added completed PIA list view

### PR DESCRIPTION
**Pull request by antsand**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Comment by binggaofw** [comment]
the text does not match design, should be completed PIAs
Screenshot 2023-07-20 at 6 47 47 PM

File: src/frontend/src/routes/router.tsx [hunk 1]
```diff
@@ -67,6 +68,7 @@ const Router = () => {
           <Route index element={<Navigate relative="path" to="list" />} />
 
           <Route path="list" element={<PIAList />} />
```
Comment (binggaofw) [inline src/frontend/src/routes/router.tsx:70]: i would recommend change the PIAList to PIAActiveList, better to understand

**Comment by binggaofw** [comment]
this file I believe you are added by accident, can be delete src/frontend/src/components/public/PIAFormTabs/review/checkAdmNeeded.tsx

File: src/frontend/src/components/public/ListViewRender/searchAndFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import SearchBox from '../../common/SearchBox';
+import PIAIntakeFilter from '../PIAIntakeFilter';
+import { IsearchAndFilter } from './interface';
+
+const SearchAndFilter = (props: IsearchAndFilter) => {
+  const showIntakeFilter = () => {
+    if (props.showfilter) {
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/searchAndFilter.tsx:7]: to make naming consistent, better to change showFilter

File: src/frontend/src/components/public/ListViewRender/searchAndFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import SearchBox from '../../common/SearchBox';
+import PIAIntakeFilter from '../PIAIntakeFilter';
+import { IsearchAndFilter } from './interface';
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/searchAndFilter.tsx:3]: ISearchAndFilter

File: src/frontend/src/components/public/ListViewRender/index.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    return filters;
+  };
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [SortBy, setSortBy] = useState('');
+  const [SortOrder, setSortOrder] = useState(0);
+  const [currentPage, setcurrentPage] = useState(1);
+  const [headings, setHeading] = useState(tableHeadingProperties);
+  const [PageSizedefault, setPageSizedefault] = useState(10);
+  const [filteMetaParams, setfilteMetaParams] = useState(showFilters);
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/index.tsx:39]:
```suggestion
  const [filterMetaParams, setFilterMetaParams] = useState(showFilters);
```

File: src/frontend/src/components/public/ListViewRender/index.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+  function filterChangeHandler(params: URLSearchParams) {
+    setcurrentPage(1);
+    setSearchParams(params);
+  }
+
+  return (
+    <div className="bcgovPageContainer background bcgovPageContainer__with-controls wrapper">
+      <div className="page__controls full__width">
+        <h1>List of PIAs</h1>
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/index.tsx:139]:
```suggestion
        <h1>Completed PIAs</h1>
```

File: src/frontend/src/components/public/ListViewRender/interface.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+export interface IfilterMetaParams {
+  [key: string]: boolean;
+}
+
+export interface IsearchAndFilter {
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/interface.ts:5]:
```suggestion
export interface ISearchAndFilter {
```

File: src/frontend/src/components/public/ListViewRender/interface.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  filterChangeHandler: (params: URLSearchParams) => void;
+  searchParams: URLSearchParams;
+  searchText: string;
+  handleSearchTextChange: (newSearchText: any) => void;
+  setSearchParamsForSearchText: () => void;
+  updateSearchUrl: () => void;
+  handleClearSearchText: () => void;
+  showfilter: IfilterMetaParams | null;
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/interface.ts:13]:
```suggestion
  showFilter: IFilterMetaParams | null;
```

File: src/frontend/src/components/public/ListViewRender/interface.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+export interface IfilterMetaParams {
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/interface.ts:1]:
```suggestion
export interface IFilterMetaParams {
```

File: src/frontend/src/components/public/ListViewRender/searchAndFilter.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+const SearchAndFilter = (props: IsearchAndFilter) => {
```
Comment (binggaofw) [inline src/frontend/src/components/public/ListViewRender/searchAndFilter.tsx:5]:
```suggestion
const SearchAndFilter = (props: ISearchAndFilter) => {
```

**Comment by binggaofw** [comment]
the search by drafter name and pia title in completed tab does not work in my local,

File: src/frontend/src/pages/PIAActiveListPage/index.tsx [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import ListViewRender from '../../components/public/ListViewRender';
+
+const PIAActiveList = () => {
+  return <ListViewRender showCompleted={false}
+  title={`List of PIA's`} />;
```
Comment (binggaofw) [inline src/frontend/src/pages/PIAActiveListPage/index.tsx:5]:
```suggestion
  title={`Active PIA's`} />;
```

**Comment by binggaofw** [comment]
the complete status pia show in active list page
Screenshot 2023-07-21 at 10 28 33 AM

File: src/frontend/src/components/common/Navbar/navPages.ts [hunk 11]
```diff
@@ -7,6 +7,12 @@ export const NavPages = [
   },
   {
     id: 2,
+    label: 'Completed PIAs',
```
Comment (binggaofw) [inline src/frontend/src/components/common/Navbar/navPages.ts:10]: the nav also need to change list of PIAs to Active PIAs

**Comment by antsand** [comment]
> the complete status pia show in active list page Screenshot 2023-07-21 at 10 28 33 AM

correct! this will be fixed in the backend.

**Comment by kushal-arora-fw** [comment]
You got Linting failure @antsand

**Pull request merged**